src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -137,6 +142,57 @@ object SonarLintIntelliJClient : SonarLintRpcClientDelegate {
         suggestionsByConfigScopeId.forEach { (configScopeId, suggestions) -> suggestAutoBind(findProject(configScopeId), suggestions) }
     }
 
+    override fun suggestConnection(suggestionsByConfigScope: MutableMap<String, MutableList<ConnectionSuggestionDto>>?) {
+        if (suggestionsByConfigScope != null && suggestionsByConfigScope.size == 1) {
+            val dto = suggestionsByConfigScope.entries.stream().findFirst().get()
+
+            val (connectionKind, projectKey, connectionName) = getAutoShareConfigParams(dto)
+
+            val project = BackendService.findModule(dto.key)?.project
+                ?: BackendService.findProject(dto.key) ?: return
+
+            ConfigurationSharing.showAutoSharedConfigurationNotification(
+                project, String.format(
+                    """
+                    A Connected Mode configuration file is available to bind to project "%s" on %s "%s".
+                """.trimIndent(), projectKey, connectionKind, connectionName
+                ), SKIP_AUTO_SHARE_CONFIGURATION_DIALOG_PROPERTY
+            )
+        }
+    }
+
+    private fun getAutoShareConfigParams(dto: MutableMap.MutableEntry<String, MutableList<ConnectionSuggestionDto>>): Triple<String, String, String> {
+        val isSQ = dto.value.stream().findFirst().get().connectionSuggestion.isLeft
+
+        var sq = SonarQubeConnectionSuggestionDto(null, null)
```

Comment:
As a general practice, I'd not recommend using DTO object for internal usage on the client side, it's should be when transferring data between back and front